torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -65,6 +65,7 @@ std::shared_ptr<NCCLComm> NCCLComm::split(
   LOG(INFO) << "Rank " << source->rank_ << ": split from parent comm "
             << source->repr() << " with color_id " << color_id << " and rank "
             << rank;
+  at::cuda::OptionalCUDAGuard gpuGuard(source->deviceIndex_);
```

Comment:
Will this fail if called with a device index of `-1`? So we would never init a comm with an uninitialized device index, correct?

Reply:
Yes, all of our created comms should have recorded a device index. If this line can fail by itself, then it is great (better than silent pass-through).